components/password.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import classNames from 'classnames';
+
+export default function Password ({
+	unknownUser = false,
+	hasError = false,
+	pattern = null,
+	isDisabled = false,
+	fieldId = 'passwordField',
+	inputId = 'password',
+	inputName = 'password',
+}) {
+
+	const PasswordFieldClassNames = classNames([
+		'o-forms o-forms--wide',
+		'ncf__field',
+		'js-field',
+		{
+			'o-forms--error': hasError,
+			'js-unknown-user-field': unknownUser
+		}
+	]);
+
+	return (
+		<div
+			id={fieldId}
+			className={PasswordFieldClassNames}
+			data-ui-item="form-field"
+			data-ui-item-name="password"
+			data-validate="required,password"
+		>
+
+			<label htmlFor="password" className="o-forms__label">Password</label>
+			<small id="password-description" className="o-forms__additional-info">
+				Use 8 or more characters with a mix of letters, numbers &amp; symbols
+			</small>
+
+			<div className="o-forms__affix-wrapper js-show-password">
+				<input
+					type="password"
+					id={inputId}
+					name={inputName}
+					placeholder="Enter a password"
+					className="no-mouseflow o-forms__text o-forms__text--suffixed js-field__input js-show-password__password-input js-item__value"
+					autoComplete="new-password"
+					data-trackable="field-password"
+					aria-describedby="password-description"
+					aria-required="true" required
+					pattern={pattern}
+					disabled={isDisabled} />
+				<span className="o-forms__suffix">
+					<input type="checkbox" id="showPassword" name="showPassword" className="o-forms__checkbox js-show-password__checkbox" data-trackable="field-show-password" />
```

Comment:
If this field will be displayed multiple times on the same page, we should make sure that the `id` and `name` for this checkbox is unique. Maybe use `inputId` to generate the `id`(?). Something like:

```suggestion
					<input type="checkbox" id=`showPassword-${inputId}` name="showPassword" className="o-forms__checkbox js-show-password__checkbox" data-trackable="field-show-password" />
```

Will need to update the `htmlFor` on the label as well.

Reply:
Great suggestion. Implementing...

Reply:
I need to do some workaround though to make it backward compatible 👍